rewrite-java/src/main/java/org/openrewrite/java/AddImport.java

```diff
@@ -151,23 +147,37 @@ public AddImport(String type, @Nullable String statik, boolean onlyIfReferenced)
                     .map(JavaSourceSet::getClasspath)
                     .orElse(Collections.emptyList());
 
-            cu = cu.getPadding().withImports(layoutStyle.addImport(cu.getPadding().getImports(), importToAdd,
-                    cu.getPackageDeclaration(), classpath));
+            List<JRightPadded<J.Import>> newImports = layoutStyle.addImport(cu.getPadding().getImports(), importToAdd, cu.getPackageDeclaration(), classpath);
+
+            // ImportLayoutStile::addImport adds always `\n` as newlines. Checking if we need to fix them
+            newImports = checkCRLF(cu, newImports);
+
+            cu = cu.getPadding().withImports(newImports);
 
             JavaSourceFile c = cu;
-            cu = cu.withClasses(ListUtils.map(cu.getClasses(), (i, clazz) -> {
-                if (i == 0) {
-                    J.ClassDeclaration cl = autoFormat(clazz, clazz.getName(), p, new Cursor(null, c));
-                    clazz = clazz.withPrefix(clazz.getPrefix().withWhitespace(cl.getPrefix().getWhitespace()));
-                }
-                return clazz;
+            cu = cu.withClasses(ListUtils.mapFirst(cu.getClasses(), clazz -> {
+                J.ClassDeclaration cl = autoFormat(clazz, clazz.getName(), p, new Cursor(null, c));
+                return clazz.withPrefix(clazz.getPrefix().withWhitespace(cl.getPrefix().getWhitespace()));
             }));
 
             j = cu;
         }
         return j;
     }
 
+    private List<JRightPadded<J.Import>> checkCRLF(JavaSourceFile cu, List<JRightPadded<J.Import>> newImports) {
+        GeneralFormatStyle generalFormatStyle = Optional.ofNullable(((SourceFile) cu).getStyle(GeneralFormatStyle.class))
+                .orElse(autodetectGeneralFormatStyle(cu));
+        if (generalFormatStyle.isUseCRLFNewLines()) {
```

Comment:
I'm wondering if we should even call `autodetectGeneralFormatStyle` here, or just skip on detecting, as I'd assume a different part of the code base should already add this style to the `JavaSourceFile`.

Reply:
it's extracted from the auto format visitor (called by autoformat). In the tests at least, there was no markers there, and autodetectGeneralFormatStyle was used.